pkg/controllers/clusterset/managedclusterlabel/managedclusterlabel_controller.go

```diff
@@ -0,0 +1,154 @@
+package managedclusterlabel
+
+import (
+	"context"
+
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+
+	utils "github.com/open-cluster-management/multicloud-operators-foundation/pkg/utils"
+	clustersetutils "github.com/open-cluster-management/multicloud-operators-foundation/pkg/utils/clusterset"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// This controller sync the managedcluster's clusterSet Label with releated clusterdeployment's ClusterSet only when the clusterdeployment is claimed from pool.
```

Comment:
shouldn't it has been handled by clusterdeployment_controller?

Reply:
clusterdeployment _controller used to add/remove clusterset label for clusterdeployment. 
this controller used to sync the clusterdeployment clusterset to managedcluster.